### PR TITLE
MagicEightBall: Converting to text template

### PR DESCRIPTION
**Pull request by mintsoft**
###### Description of changes

Ref: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2767

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/magic_eight_ball

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @jlbaez

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/MagicEightBall.t [hunk 1]
```diff
@@ -11,34 +11,25 @@ zci is_cached   => 0;
 
 my $possibleAnswers = re(qr/(It is certain|It is decidedly so|Without a doubt|Yes, definitely|You may rely on it|As I see it, yes|Most likely|Outlook good|Yes|Signs point to yes|Reply hazy try again|Ask again later|Better not tell you now|Cannot predict now|Concentrate and ask again|Don't count on it|My reply is no|My sources say no|Outlook not so good|Very doubtful)/);
 
-my $possibleStructuredAnswer = re(qr/^(?:It is certain|It is decidedly so|Without a doubt|Yes, definitely|You may rely on it|As I see it, yes|Most likely|Outlook good|Yes|Signs point to yes|Reply hazy try again|Ask again later|Better not tell you now|Cannot predict now|Concentrate and ask again|Don&#39;t count on it|My reply is no|My sources say no|Outlook not so good|Very doubtful)$/);
+sub build_test
+{
+    my ($text, $input) = @_;
```
Comment (GuiltyDolphin) [inline t/MagicEightBall.t]: We only need `$input`, and in the tests below we only need to pass in the query (seeing as we are just using the same response regex).

File: t/MagicEightBall.t [hunk 2]
```diff
@@ -11,34 +11,25 @@ zci is_cached   => 0;
 
 my $possibleAnswers = re(qr/(It is certain|It is decidedly so|Without a doubt|Yes, definitely|You may rely on it|As I see it, yes|Most likely|Outlook good|Yes|Signs point to yes|Reply hazy try again|Ask again later|Better not tell you now|Cannot predict now|Concentrate and ask again|Don't count on it|My reply is no|My sources say no|Outlook not so good|Very doubtful)/);
 
-my $possibleStructuredAnswer = re(qr/^(?:It is certain|It is decidedly so|Without a doubt|Yes, definitely|You may rely on it|As I see it, yes|Most likely|Outlook good|Yes|Signs point to yes|Reply hazy try again|Ask again later|Better not tell you now|Cannot predict now|Concentrate and ask again|Don&#39;t count on it|My reply is no|My sources say no|Outlook not so good|Very doubtful)$/);
+sub build_test
+{
```
Comment (GuiltyDolphin) [inline t/MagicEightBall.t]: @mintsoft Cuddles!

**Comment by GuiltyDolphin** [comment]
@mintsoft Thanks! :dolphin:

**Pull request merged**